src/FindApprenticeshipTraining/SFA.DAS.FindApprenticeshipTraining/Application/TrainingCourses/Queries/GetTrainingCourseProvider/GetTrainingCourseProviderQueryHandler.cs

```diff
@@ -1,134 +1,162 @@
+            var ukprnsCountTask = _roatpV2ApiClient.Get<GetTotalProvidersForStandardResponse>(
+                new GetTotalProvidersForStandardRequest(request.CourseId));
+
+            var providerCoursesTask = _roatpV2ApiClient.Get<List<GetProviderAdditionalStandardsItem>>(
                 new GetProviderAdditionalStandardsRequest(request.ProviderId));
-            var overallAchievementRatesTask = _courseDeliveryApiClient.Get<GetOverallAchievementRateResponse>(
+
+            var overallAchievementRatesTask = _roatpV2ApiClient.Get<GetOverallAchievementRateResponse>(
                 new GetOverallAchievementRateRequest(courseTask.Result.SectorSubjectAreaTier2Description));
 
             var apprenticeFeedbackTask = _apprenticeFeedbackApiClient.GetWithResponseCode<GetApprenticeFeedbackResponse>(new GetApprenticeFeedbackDetailsRequest(request.ProviderId));
             var employerFeedbackTask = _employerFeedbackApiClient.GetWithResponseCode<GetEmployerFeedbackResponse>(new GetEmployerFeedbackDetailsRequest(request.ProviderId));
 
-            var coursesTask = _cacheHelper.GetRequest<GetStandardsListResponse>(_coursesApiClient,
-                new GetAvailableToStartStandardsListRequest(), nameof(GetStandardsListResponse), out var saveToCache);
-
-            var shortlistTask = _shortlistService.GetShortlistItemCount(request.ShortlistUserId);
-            
-            await Task.WhenAll(providerTask, coursesTask, providerCoursesTask, ukprnsCount, overallAchievementRatesTask, shortlistTask, apprenticeFeedbackTask, employerFeedbackTask);
+            var shortlistTask = request.ShortlistUserId.HasValue
+                ? _shortlistApiClient.Get<List<ShortlistItem>>(new GetShortlistForUserRequest(request.ShortlistUserId.Value))
+                : Task.FromResult(new List<ShortlistItem>());
 
-            if (providerTask.Result == null && locationTask.Result != null)
-            {
-                providerTask = Task.FromResult(
-                    await _courseDeliveryApiClient.Get<GetProviderStandardItem>(
-                        new GetProviderByCourseAndUkPrnRequest(request.ProviderId, request.CourseId, courseTask.Result.SectorSubjectAreaTier2Description, null, null, request.ShortlistUserId)));
+            await Task.WhenAll(providerCoursesTask, ukprnsCountTask, overallAchievementRatesTask, shortlistTask, apprenticeFeedbackTask, employerFeedbackTask);
 
-                if (providerTask.Result != null)
-                {
-                    providerTask.Result.DeliveryTypes = new List<GetDeliveryTypeItem> 
-                    {
-                        new GetDeliveryTypeItem
-                        {
-                            DeliveryModes = "NotFound"
-                        } 
-                    };    
-                }
-            }
+            var providerDetails = await GetProviderDetails(request.ProviderId, request.CourseId, locationTask.Result, shortlistTask.Result);
```

Comment:
This can be included in the parallel task as well.

Reply:
No it cannot, as it requires the result from shortlistTask